apps/dashboard/src/components/layouts/layout-editor-provider.tsx

```diff
@@ -78,11 +80,49 @@ export const LayoutEditorProvider = ({ children }: { children: React.ReactNode }
     });
   }, 500);
 
+  const setFormIssues = useCallback(
+    (controlIssues?: Record<string, RuntimeIssue[]>) => {
+      const flattenedIssues = flattenIssues(controlIssues);
+      const layoutIssues = Object.keys(flattenedIssues).reduce(
+        (acc, key) => {
+          acc[key.replace('email.', '')] = flattenedIssues[key];
+          return acc;
+        },
+        {} as Record<string, string>
+      );
+
+      const currentErrors = form.formState.errors;
+      Object.keys(currentErrors).forEach((key) => {
+        if (!layoutIssues[key]) {
+          form.clearErrors(key);
+        }
+      });
+
+      Object.entries(layoutIssues).forEach(([key, value]) => {
+        form.setError(key as string, { message: value });
+      });
+    },
+    [form]
+  );
+
   const { updateLayout, isPending: isUpdating } = useUpdateLayout({
     onSuccess: () => {
       showSuccessToast('Layout updated successfully', '', toastOptions);
     },
     onError: (error) => {
+      if (error instanceof NovuApiError && 'controls' in (error.rawError as any)) {
+        const controlIssues = (error.rawError as any).controls;
+        setFormIssues(controlIssues);
+
+        const firstControlError = getFirstErrorMessage({ controls: controlIssues }, 'controls');
+        showErrorToast(
+          firstControlError?.message ?? 'Failed to update layout',
+          'Failed to update layout',
+          toastOptions
+        );
+        return;
+      }
```

Comment:
when got bad request then set the form issues and show the toast with the first error